controllers/fdb_client.go

```diff
@@ -1,110 +1,78 @@
-	}
+// DatabaseClientProvider provides an abstraction for creating clients that
+// communicate with the database.
+type DatabaseClientProvider interface {
+	// GetLockClient generates a client for working with locks through the database.
+	GetLockClient(cluster *fdbtypes.FoundationDBCluster) (LockClient, error)
 
-	database, err = fdb.OpenDatabase(clusterFilePath)
-	if err != nil {
-		return fdb.Database{}, err
-	}
+	// GetAdminClient generates a client for performing administrative actions
+	// against the database.
+	GetAdminClient(cluster *fdbtypes.FoundationDBCluster, kubernetesClient client.Client) (AdminClient, error)
 
-	err = database.Options().SetTransactionTimeout(defaultTransactionTimeout)
-	if err != nil {
-		return fdb.Database{}, err
-	}
-
-	fdbDatabaseCache[cacheKey] = database
-	return database, nil
+	// CleanUpCache removes the cache entry for a cluster.
+	CleanUpCache(namespace string, name string)
 }
 
-// cleanUpDBCache removes the FDB DB connection for the deleted cluster.
-// Otherwise the controller will connect to an old cluster.
-func cleanUpDBCache(namespace string, name string) {
-	delete(fdbDatabaseCache, fmt.Sprintf("%s/%s", namespace, name))
+type legacyDatabaseClientProvider struct {
+	AdminClientProvider func(cluster *fdbtypes.FoundationDBCluster, kubernetesClient client.Client) (AdminClient, error)
+	LockClientProvider  LockClientProvider
 }
 
-// getStatusFromDB gets the database's status directly from the system key
-func getStatusFromDB(cluster *fdbtypes.FoundationDBCluster) (*fdbtypes.FoundationDBStatus, error) {
-	log.Info("Fetch status from FDB", "namespace", cluster.Namespace, "cluster", cluster.Name)
-	statusKey := "\xff\xff/status/json"
-
-	database, err := getFDBDatabase(cluster)
-	if err != nil {
-		return nil, err
-	}
+// GetLockClient generates a client for working with locks through the database.
+func (p legacyDatabaseClientProvider) GetLockClient(cluster *fdbtypes.FoundationDBCluster) (LockClient, error) {
+	return p.LockClientProvider(cluster)
+}
 
-	result, err := database.Transact(func(transaction fdb.Transaction) (interface{}, error) {
-		err := transaction.Options().SetAccessSystemKeys()
-		if err != nil {
-			return nil, err
-		}
-		// Wait default timeout seconds to receive status for larger clusters.
-		err = transaction.Options().SetTimeout(int64(DefaultCLITimeout * 1000))
-		if err != nil {
-			return nil, err
-		}
+// GetAdminClient generates a client for performing administrative actions
+// against the database.
+func (p legacyDatabaseClientProvider) GetAdminClient(cluster *fdbtypes.FoundationDBCluster, kubernetesClient client.Client) (AdminClient, error) {
+	return p.AdminClientProvider(cluster, kubernetesClient)
+}
 
-		statusBytes := transaction.Get(fdb.Key(statusKey)).MustGet()
-		if len(statusBytes) == 0 {
-			return nil, err
-		}
+// CleanUpCache removes the cache entry for a cluster.
+func (p legacyDatabaseClientProvider) CleanUpCache(namespace string, name string) {
+}
```

Comment:
It seems like the implementation is missing here?

Reply:
I don't think we have a way to expose the cache clearing behavior, so we have a dummy implementation instead.